lib/migrator/helpers.ts

```diff
@@ -198,3 +199,77 @@ export const getTargetBlockDevice = async (mountLabel: string = 'C') => {
 		keepOriginal: true
 	})
 }
+
+/** 
+ * Identifies the type of WiFi authentication for a connection profile.
+ * NONE means no authentication.
+ */
+export const enum WifiAuthType { NONE, WPA2_PSK, WPA3_SAE }
+
+/** Configuration values for a single WiFi network profile. */
+export interface ConnectionProfile {
+	name: string;
+	wifiSsid: string;
+	wifiAuthType: WifiAuthType;
+	wifiKey: string;
+}
+
+/** Generates the contents for a balenaOS network configuration file. */
```

Comment:
What's about this function that makes it balenaOS-specific?

Reply:
Yeah, I think it’s regular NM config, though please confirm with the OS team.

My larger point though is that anytime you feel the code is closely tied with balena, it’s a signal that the code is not general enough and you should reach out. Etcher is motivated by balena’s use cases but its APIs must be usable by third-parties for it to ever become extensible.

Reply:
Got confirmation that the configuration file is pure NetworkManager.